### DISCUSSION TITLE
How do you differentiate between no hits return vs loading (fetching from network) state?

### DISCUSSION THREAD

**Discussion by landermedia** [algolia/instantsearch#5358] [Q&A] [answered]
Using `useInfiniteHits` return you `hits` but it can sometimes return empty result depending on the current search query or even data in your indice. 

Our UI as of the moment sees no difference between fetching from network state vs no results state from response. We'd like to show proper loading indicators when the hook is currently fetching from network and only show really something else for empty results returned. How do we go about this?

EDIT: This is also true for other hooks like `useHierarchicalMenu`, `useRefinementList`, etc.

**Comment by Haroenv** [comment]
`useSearchBox` (only for now) has a stalled search boolean you can use as an indicator that will be marked as "stalled" after 200ms of unresolved network activity. You can use it like this:

```js
function Loader() {
  const { isSearchStalled } = useSearchBox({}, { $$widgetType: "custom.loadingIndicator" });

  if (isSearchStalled) {
    return "search loading…";
  }
  return null
}
```

**Reply by landermedia** [reply]
@Haroenv thanks for the prompt response. I've tried this but I can't get `isSearchStalled` to transition from `true` to `false`. Not sure what I'm missing, hmm. 🤔

<img width="773" alt="image" src="https://user-images.githubusercontent.com/83575285/164713435-a61ccd87-4aad-4c5b-b9bd-c86106ae8176.png">

**Reply by Haroenv** [reply]
Stalled is by default defined as 200ms of network inactivity, so you can lower your network speed to see it more obvious

**Comment by dhayab** [comment, marked as answer]
Hi, you can now leverage [`useInstantSearch()`](https://www.algolia.com/doc/api-reference/widgets/use-instantsearch/react-hooks/#widget-param-status) to retrieve the `status` of your search implementation.

**Reply by NicolasGorga** [reply]
Hello @dhayab i am seeing that the status is idle a lot, before actually passing to loading and idle again with results. Since i am checking that if the status is idle and items.length <= 0 to show a "No results text", this is popping up, then showing the actual results which is not desirable. Am i missing something?

**Reply by Haroenv** [reply]
do you have a reproduction for this? please open an issue with clear instructions on how to see it happening :)
